### PR TITLE
[KIECLOUD-19] Support additional environments, start w/ prod

### PR DESCRIPTION
**Pull request by tchughesiv**
[KIECLOUD-19] Support additional environments, start w/ prod

 - made it easier to add new envs/objects going forward
 - operator now automatically sets apiversion and kind for all objects
 - added prod environment

Signed-off-by: tchughesiv <tchughesiv@gmail.com>

### CODE REVIEW

File: internal/app/handler/handler_test.go [hunk 1]
```diff
@@ -1,66 +1,56 @@
...
 		},
 		Deleted: false}
-	logrus.Debugf("Testing with environment %v", event.Object.(*v1.App).Spec.Environment)
+	logrus.Debugf("Testing with environment %v", event.Object.(*opv1.App).Spec.Environment)
 
 	defer func() {
 		err := recover().(error)
 		logrus.Debugf("Failed with error %v", err)
-		assert.Equal(t, err.Error(), "lookup kubernetes.default.svc: no such host", "Did not get expected no such host error")
+		assert.Equal(t, "lookup kubernetes.default.svc on 10.19.115.218:53: no such host", err.Error(), "Did not get expected no such host error")
```
Comment (bmozaffa) [inline internal/app/handler/handler_test.go]: Was this IP address left here accidentally?

File: cmd/kie-cloud-operator/main.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func main() {
 	if err != nil {
 		logrus.Fatalf("Failed to get watch namespace: %v", err)
 	}
-	resyncPeriod := time.Duration(5) * time.Second
+	resyncPeriod := time.Duration(2) * time.Second
```
[the remark this answers is not in the document]
Comment (tchughesiv) [reply cmd/kie-cloud-operator/main.go]: Based on the code I've looked at, this number is how often the Operator will react to an events. I figured now that we have status and logging working more cleanly, why not have the operator react more quickly?

File: internal/app/handler/handler_test.go (hunk 1 above)
Comment (tchughesiv) [reply internal/app/handler/handler_test.go]: ya, also this test needs work so its actually commented out in the code. I'm not sure its a valid test.

File: cmd/kie-cloud-operator/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmozaffa) [reply cmd/kie-cloud-operator/main.go]: Not sure, maybe... we really need some best practices around this kind of thing, and to know if there is a meaningful performance hit.

File: pkg/apis/kiegroup/v1/types.go [hunk 3]
```diff
@@ -31,12 +32,16 @@ type AppSpec struct {
...
-type OpenShiftObject struct {
-	DeploymentConfig apiv1.DeploymentConfig `json:"deployment"`
-	Service          corev1.Service         `json:"service"`
-	Route            routev1.Route          `json:"route"`
+type CustomObject struct {
+	PersistentVolumeClaims []corev1.PersistentVolumeClaim `json:"persistentvolumeclaims,omitempty"`
+	ServiceAccounts        []corev1.ServiceAccount        `json:"serviceaccounts,omitempty"`
+	Secrets                []corev1.Secret                `json:"secrets,omitempty"`
+	RoleBindings           []authv1.RoleBinding           `json:"rolebindings,omitempty"`
+	DeploymentConfigs      []appsv1.DeploymentConfig      `json:"deploymentconfigs,omitempty"`
```
Comment (bmozaffa) [inline pkg/apis/kiegroup/v1/types.go]: When would a single object have multiple DCs?

File: internal/app/handler/handler_test.go (hunk 1 above)
Comment (tchughesiv) [reply internal/app/handler/handler_test.go]: wait... this looks wrong... i need to fix it.

File: pkg/apis/kiegroup/v1/types.go (hunk 3 above)
Comment (tchughesiv) [reply pkg/apis/kiegroup/v1/types.go]: w/ a prod deployment, the kie servers also require postgres... which has its own dc.

File: cmd/kie-cloud-operator/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tchughesiv) [reply cmd/kie-cloud-operator/main.go]: returned to 5 for now

File: internal/app/handler/handler_test.go (hunk 1 above)
Comment (tchughesiv) [reply internal/app/handler/handler_test.go]: fixed

**Pull request merged**